xmtp_mls/src/groups/group_permissions.rs

```diff
@@ -1,162 +1,247 @@
+};
+
+// A trait for policies that can add/remove members and installations for the group
+pub trait MembershipPolicy: std::fmt::Debug {
     fn evaluate(&self, actor: &CommitParticipant, change: &AggregatedMembershipChange) -> bool;
+    fn to_proto(&self) -> MembershipPolicyProto;
+}
+
+#[derive(Debug, Error)]
+pub enum PolicyError {
+    #[error("serialization {0}")]
+    Serialization(#[from] prost::EncodeError),
+    #[error("deserialization {0}")]
+    Deserialization(#[from] prost::DecodeError),
+    #[error("invalid policy")]
+    InvalidPolicy,
 }
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, PartialEq)]
 #[allow(dead_code)]
-pub enum StandardPolicies {
+#[repr(u8)]
+pub enum BasePolicies {
     Allow,
     Deny,
     // Allow if the change only applies to subject installations with the same account address as the actor
     AllowSameMember,
+    // Allow the change if the actor is the creator of the group
+    AllowIfActorCreator,
     // AllowIfActorAdmin, TODO: Enable this once we have admin roles
-    // AllowIfActorCreator, TODO: Enable this once we know who the creator is
     // AllowIfSubjectRevoked, TODO: Enable this once we have revocation and have context on who is revoked
 }
 
-impl Policy for StandardPolicies {
+impl MembershipPolicy for BasePolicies {
     fn evaluate(&self, actor: &CommitParticipant, change: &AggregatedMembershipChange) -> bool {
         match self {
-            StandardPolicies::Allow => true,
-            StandardPolicies::Deny => false,
-            StandardPolicies::AllowSameMember => change.account_address == actor.account_address,
+            BasePolicies::Allow => true,
+            BasePolicies::Deny => false,
+            BasePolicies::AllowSameMember => change.account_address == actor.account_address,
+            BasePolicies::AllowIfActorCreator => true, // TODO: Enable proper check once we can tell who the creator is
         }
     }
-}
 
-// An AndCondition evaluates to true if all the policies it contains evaluate to true
-#[derive(Clone, Debug)]
-pub struct AndCondition<PolicyType>
-where
-    PolicyType: Policy,
-{
-    policies: Vec<PolicyType>,
+    fn to_proto(&self) -> MembershipPolicyProto {
+        let inner = match self {
+            BasePolicies::Allow => BasePolicyProto::Allow as i32,
+            BasePolicies::Deny => BasePolicyProto::Deny as i32,
+            BasePolicies::AllowSameMember => BasePolicyProto::AllowSameMember as i32,
+            BasePolicies::AllowIfActorCreator => BasePolicyProto::AllowIfActorCreator as i32,
+        };
+
+        MembershipPolicyProto {
+            kind: Some(PolicyKindProto::Base(inner)),
+        }
+    }
 }
 
+#[derive(Debug, Clone, PartialEq)]
 #[allow(dead_code)]
-impl<PolicyType> AndCondition<PolicyType>
-where
-    PolicyType: Policy,
-{
-    pub fn new(policies: Vec<PolicyType>) -> Self {
-        Self { policies }
+pub enum MembershipPolicies {
```

Comment:
Not super important, but could `MembershipPolicies` be a trait on `BasePolicies`, `AndCondition` and `AnyCondition`, removing the need for this enum and the match arms?

Reply:
That's how it started before this PR. Things got really nasty trying to convert the proto oneof value into one of several possible `dyn Policy` types.